--- hawknl/src/makefile.linux.orig	2003-12-30 09:06:00.000000000 +0000
+++ hawknl/src/makefile.linux	2007-09-29 09:55:18.000000000 +0000
@@ -11,9 +11,9 @@
 OUTPUT = libNL.so.$(VERSION)
 LIBNAME = NL
 STATIC = libNL.a
-OPTFLAGS = -funroll-all-loops -ffast-math -fomit-frame-pointer -O2 -D_GNU_SOURCE -D_REENTRANT
-CFLAGS = -Wall -fPIC $(INCLUDE) $(OPTFLAGS)
-LIBFLAGS = -shared -Wl,-soname,NL.so.$(MAJOR_VERSION).$(MINOR_VERSION) -rdynamic -lpthread
+OPTFLAGS = -funroll-all-loops -ffast-math -fomit-frame-pointer -O2
+CFLAGS = -Wall -fPIC $(INCLUDE) $(OPTFLAGS) -D_GNU_SOURCE -D_REENTRANT
+LIBFLAGS = -shared -Wl,-soname,libNL.so.$(MAJOR_VERSION).$(MINOR_VERSION) -rdynamic -lpthread
 OBJECTS = crc.o errorstr.o nl.o sock.o group.o loopback.o err.o thread.o mutex.o condition.o nltime.o
 
 all: $(OBJECTS)
@@ -34,15 +34,13 @@
 nltime.o : nlinternal.h nltime.c
 
 install:
-	cp $(OUTPUT) $(LIBDIR)
-	cp $(STATIC) $(LIBDIR)
-	chmod 755 $(LIBDIR)/$(OUTPUT)
-	ln -s $(LIBDIR)/$(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
-	ln -s $(LIBDIR)/$(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION)
-	ln -s $(LIBDIR)/$(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so
-	ln -s $(LIBDIR)/$(OUTPUT) $(LIBDIR)/$(LIBNAME).so
-	cp ../include/nl.h $(INCDIR)/nl.h
-	chmod 644 $(INCDIR)/nl.h
+	cp $(OUTPUT) $(DESTDIR)/$(LIBDIR)
+	cp $(STATIC) $(DESTDIR)/$(LIBDIR)
+	chmod 755 $(DESTDIR)/$(LIBDIR)/$(OUTPUT)
+	ln -sf $(OUTPUT) $(DESTDIR)/$(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
+	ln -sf $(OUTPUT) $(DESTDIR)/$(LIBDIR)/lib$(LIBNAME).so
+	cp ../include/nl.h $(DESTDIR)/$(INCDIR)/nl.h
+	chmod 644 $(DESTDIR)/$(INCDIR)/nl.h
 	@echo ""
 	@echo "*************************************************"
 	@echo "* Installed HawkNL.                             *"
@@ -53,12 +53,12 @@
 	@echo "*************************************************"
 
 uninstall:
-	rm -f $(LIBDIR)/$(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
-	rm -f $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
-	rm -f $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION)
-	rm -f $(LIBDIR)/lib$(LIBNAME).so
-	rm -f $(LIBDIR)/$(STATIC)
-	rm -f $(INCDIR)/nl.h
+	rm -f $(DESTDIR)/$(LIBDIR)/$(OUTPUT) $(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
+	rm -f $(DESTDIR)/$(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION).$(MINOR_VERSION)
+	rm -f $(DESTDIR)/$(LIBDIR)/lib$(LIBNAME).so.$(MAJOR_VERSION)
+	rm -f $(DESTDIR)/$(LIBDIR)/lib$(LIBNAME).so
+	rm -f $(DESTDIR)/$(LIBDIR)/$(STATIC)
+	rm -f $(DESTDIR)/$(INCDIR)/nl.h
 
 .PHONY : clean
 clean:
